packages/s/sqlgen/xmake.lua

```diff
@@ -0,0 +1,94 @@
+package("sqlgen")
+    set_homepage("https://github.com/getml/sqlgen")
+    set_description("sqlgen is an ORM and SQL query generator for C++-20, similar to Python's SQLAlchemy/SQLModel or Rust's Diesel.")
+    set_license("MIT")
+
+    add_urls("https://github.com/getml/sqlgen/archive/refs/tags/$(version).tar.gz",
+             "https://github.com/getml/sqlgen.git")
+
+    add_versions("v0.2.0", "c093036ebdf2aaf1003b2d1623713b97106ed43b1d39dc3d4f38e381f371799e")
+
+    add_patches("0.2.0", "patches/0.2.0/cmake.patch", "e9819b9a8a2c8f8a5b6c553eac3bb10fc65856aa9af451f83e2dbf55ca6c66c0")
+
+    add_deps("cmake", "reflect-cpp")
+
+    add_configs("mysql", {description = "Enable MySQL Support", default = false, type = "boolean", readonly = true})
+    add_configs("postgres", {description = "Enable PostgreSQL Support", default = true})
+    add_configs("sqlite", {description = "Enable SQLite Support", default = true})
+
+    if is_plat("windows") then
+        add_configs("shared", {description = "Build shared library.", default = false, type = "boolean", readonly = true})
+    end
+
+    on_load(function (package)
+        if package:config("mysql") then
+            package:add("deps", "mariadb-connector-c")
+        end
+        if package:config("postgres") then
+            package:add("deps", "libpq")
+        end
+        if package:config("sqlite") then
+            package:add("deps", "sqlite3")
+        end
+    end)
+
+    on_install("windows|!arm64 or macosx|!arm64 or linux|!arm64 or bsd|!arm64", function (package)
```

Comment:
you can use on_check to simplify it, we can check arm64 in it.

```lua
on_check(function (package)
    assert(not package:is_arch("arm64"), "package(%s) does not support arm64", package:name())
end)

on_install("windows", "macosx", "linux", "bsd", ...)
```

Reply:
Since the restriction comes from the libpq dependency (from which I copied this line), is it possible to just tell xmake to check for libpq availability instead (without failing the builds)?

Reply:
you should check `package:config("postgres")` in on_check